src/control/server/config/server.go

```diff
@@ -433,8 +434,99 @@ func getAccessPointAddrWithPort(log logging.Logger, addr string, portDefault int
 	return addr, nil
 }
 
+// Calculate hugepages based on total target count if not using nvme.
+func hugepageSetCount(log logging.Logger, cfgTargetCount, sysXSCount, hugepageSize int, cfg *Server) error {
+	if cfgTargetCount <= 0 {
+		return nil // no nvme, no hugepages required
+	}
+
+	if cfg.DisableHugepages {
+		return FaultConfigHugepagesDisabled
+	}
+
+	// Calculate minimum number of hugepages for all configured engines.
+	minHugepages, err := common.CalcMinHugepages(hugepageSize, cfgTargetCount+sysXSCount)
+	if err != nil {
+		return err
+	}
+
+	// If the config doesn't specify hugepages, use the minimum. Otherwise, validate
+	// that the configured amount is sufficient.
+	if cfg.NrHugepages == 0 {
+		var msgSysXS string
+		if sysXSCount > 0 {
+			msgSysXS = fmt.Sprintf(" and %d sys-xstreams", sysXSCount)
+		}
+		log.Debugf("calculated nr_hugepages: %d for %d targets%s", minHugepages,
+			cfgTargetCount, msgSysXS)
+		cfg.NrHugepages = minHugepages
```

Comment:
This is a little weird. It would be better to either convert this function into method on `config.Server` or else return the calculated hugepages so that it can be set in the config by the caller. Doing it this way as a side-effect is harder to test and generally more error-prone, IMO.

Reply:
Done in the second batch of code review changes.